### PR TITLE
add missing binary operators to lalrpop and ast

### PR DESCRIPTION
**Pull request by rutefig**
Added missing logic, arithmetic and bitwise operators to lalrop and ast.

Covers #205 and #216 issues

### CODE REVIEW

File: src/parser/chiquito.lalrpop [hunk 1]
```diff
@@ -136,21 +136,52 @@ ParseEq: Expression<BigInt, Identifier> = {
...
+ParseBitAnd: Expression<BigInt, Identifier> =  ParseLogichBinOp<"&", ParseArith>;
+
 // Arithmetical expressions
 ParseArith: Expression<BigInt, Identifier> = {
-    ParseSum
+    ParseLeftShift
 }
 
-ParseSum: Expression<BigInt, Identifier> = ParseArithBinOp<"+", ParseMul>;
+ParseLeftShift: Expression<BigInt, Identifier> = ParseArithBinOp<"<<", ParseRightShift>;
```
Comment (leolara) [inline src/parser/chiquito.lalrpop:158]: @rutefig << and >> have the same presence so you need to group them like in ParseEq

File: src/parser/chiquito.lalrpop [hunk 2]
```diff
@@ -136,21 +136,52 @@ ParseEq: Expression<BigInt, Identifier> = {
...
+
+ParseRightShift: Expression<BigInt, Identifier> = ParseArithBinOp<">>", ParseSub>;
```
Comment (leolara) [inline src/parser/chiquito.lalrpop:160]: @rutefig group + and -

File: src/parser/chiquito.lalrpop [hunk 3]
```diff
@@ -136,21 +136,52 @@ ParseEq: Expression<BigInt, Identifier> = {
...
+
+ParseSub: Expression<BigInt, Identifier> = ParseArithBinOp<"-", ParseSum>;
+
+ParseSum: Expression<BigInt, Identifier> = ParseArithBinOp<"+", ParseDivRem>;
 
-ParseMul: Expression<BigInt, Identifier> = ParseArithBinOp<"*", ParsePrefixMinus>;
+ParseDivRem: Expression<BigInt, Identifier> = ParseArithBinOp<"%", ParseDiv>;
```
Comment (leolara) [inline src/parser/chiquito.lalrpop:166]: @rutefig group * / %, \\

**Comment by leolara** [comment]
The circom documentation references the rust documentation about precendence:
https://docs.circom.io/circom-language/basic-operators/
https://doc.rust-lang.org/1.22.1/reference/expressions/operator-expr.html#operator-precedence

This does not explain the precedence of **, ~ so we need to investigate which is actually in circom

**Comment by rutefig** [comment]
> The circom documentation references the rust documentation about precendence: https://docs.circom.io/circom-language/basic-operators/ https://doc.rust-lang.org/1.22.1/reference/expressions/operator-expr.html#operator-precedence
> 
> This does not explain the precedence of **, ~ so we need to investigate which is actually in circom

Yeah I checked the lalrop file, `~` is a unary operator so it has the same precedence as the rest of unary operators, and pow `**` comes before the mul and div group, right after the unary operators.

Regarding the groups I agree it should all be grouped :) gonna do it right away

**Review by leolara — APPROVED** [review]

**Pull request merged**